File: C#/CursoMicrosoftC#/CursoSourceProfessor/Curso2541/Modulo04/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using System.IO;

namespace Modulo04
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private DataSet Ds;

        private void Form1_Load(object sender, EventArgs e)
        {
            DataTable DtClientes = new DataTable("Cliente");

            DataColumn DcId = new DataColumn("Id", typeof(int));
            DataColumn DcNome = new DataColumn("Nome", typeof(String));

            DtClientes.Columns.Add(DcId);
            DtClientes.Columns.Add(DcNome);

            Ds = new DataSet("Clientes");
            Ds.Tables.Add(DtClientes);

            Dtg_Clientes.DataSource = Ds.Tables[0];
        }

        private void Cmd_Incluir_Click(object sender, EventArgs e)
        {
            DataRow Row = Ds.Tables[0].NewRow();
            Row["Id"] = Txt_Id.Text;
            Row["Nome"] = Txt_Nome.Text;
            Ds.Tables[0].Rows.Add(Row);
        }

        private void Cmd_GerarXml_Click(object sender, EventArgs e)
        {
            Ds.WriteXml(@"C:\Clientes.xml");
            MessageBox.Show("Xml gerado com sucesso");
        }

        private void Cmd_LerXml_Click(object sender, EventArgs e)
        {
            Cliente ObjCliente = new Cliente();
            string Xml = ObjCliente.ObterClientesXml();

            StringReader ObjStringReader = new StringReader(Xml);
            Ds.ReadXml(ObjStringReader);

            //Criação de um formulário dinâmico
            Form ObjForm = new Form();
            DataGridView ObjDataGridView = new DataGridView();
            ObjDataGridView.DataSource = Ds.Tables[0];
            ObjForm.Controls.Add(ObjDataGridView);
            ObjDataGridView.Dock = DockStyle.Fill;
            ObjForm.Show();
        }
    }
}

File: C#/CursoMicrosoftC#/CursoSourceProfessor/Curso2541VB/Modulo04/Form1.vb
Imports System.IO

Public Class Form1

    Private Ds As DataSet

    Private Sub Form1_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        Dim DtClientes As New DataTable("Cliente")

        Dim DcId As New DataColumn("Id", GetType(Integer))
        Dim DcNome As New DataColumn("Nome", GetType(String))

        DtClientes.Columns.Add(DcId)
        DtClientes.Columns.Add(DcNome)

        Ds = New DataSet("Clientes")
        Ds.Tables.Add(DtClientes)

        Dtg_Clientes.DataSource = Ds.Tables(0)
    End Sub

    Private Sub Cmd_Incluir_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Cmd_Incluir.Click
        Dim Row As DataRow = Ds.Tables(0).NewRow()
        Row("Id") = Txt_Id.Text
        Row("Nome") = Txt_Nome.Text
        Ds.Tables(0).Rows.Add(Row)
    End Sub

    Private Sub Cmd_GerarXml_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Cmd_GerarXml.Click
        Ds.WriteXml("C:\Clientes.xml")
        MessageBox.Show("Xml gerado com sucesso")
    End Sub

    Private Sub Cmd_LerXml_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Cmd_LerXml.Click
        Dim ObjCliente As New Cliente
        Dim Xml As String = ObjCliente.ObterClientesXml()

        Dim ObjStringReader As New StringReader(Xml)
        Ds.ReadXml(ObjStringReader)

        'Criação de um formulário dinâmico
        Dim ObjForm As New Form()
        Dim ObjDataGridView As New DataGridView()
        ObjDataGridView.DataSource = Ds.Tables(0)
        ObjForm.Controls.Add(ObjDataGridView)
        ObjDataGridView.Dock = DockStyle.Fill
        ObjForm.Show()
    End Sub
End Class
